lib/migration-chunks/validation/duplicate-props.js

```diff
@@ -0,0 +1,52 @@
+const errorMessages = require('./errors');
+
+const isSameEntity = (step, otherStep) => {
+  return (
+    step.payload.contentTypeId === otherStep.payload.contentTypeId &&
+    step.payload.fieldId === otherStep.payload.fieldId
+  );
+};
+
+const hasProps = (step) => step.payload && step.payload.props;
+
+const getErrorMsgParams = (step) => {
+  const type = step.type.split('/')[0];
+  const params = {
+    field: {
+      type: 'field',
+      id: step.payload.fieldId
+    },
+    contentType: {
+      type: 'content type',
+      id: step.payload.contentTypeId
+    }
+  };
+  return params[type];
+};
+
+module.exports = (chunks) => {
+  return chunks.reduce((errors, chunk) => {
+    const chunkErrors = chunk.reduce((chunkErrors, step, i) => {
+      if (hasProps(step)) {
+        const thisStepsProp = Object.keys(step.payload.props)[0];
+        const duplicate = chunk.slice(0, i).find((stepBefore) => {
```

Comment:
Can we pull this into `hasDuplicate(previousSteps, step)` to remove one nesting level?